Fix for import bug

cccs-jh opened the pull request:
Fixes #40
Importing vipermonkey fails because Context is not defined in vipermonkey.api.
I assume it used to be imported by one of the wildcard imports, this bug doesn't happen with the upstream decalage2 ViperMonkey.
importing Context explicitly from vba_context fixes the issue.
Using "import vba_context" rather than "from vba_context import Context" to avoid namespace overlap.

decalage2 committed:
setup.py/requirements.txt: regex is not required if platform is PyPy on Windows

decalage2 committed:
Merge pull request #109 from kirk-sayre-work/master

ViperMonkey update April 2021

decalage2 committed:
bumped version to 1.0.3

decalage2 committed:
setup.py, requirements: changed oletools version to 0.56.1 to fix PyPy compatibility on Windows, fixes #92, fixes #106

cccs-jh committed:
unidecode 1.3.0 dropped python2 support

cccs-jh committed:
Merge remote-tracking branch 'kirk/master' into kirk

cccs-jh committed:
Fix Context not defined in api.

Some import change caused Context to not be imported from vba_context.
Import explicitly and avoid namespace confusion

gdesmar commented:
Hello @kirk-sayre-work,

Are there any plans to merge pull requests for python2 before the python3 version is finished?

I am asking because it is starting to become a little complicated to manage the multiple forks. Yesterday I fixed one situation where multiple ADODB.Stream would be used, and their content not cleared between re-opening. It is now in PR#43 along with other fixes by the PR opener. I could make my own PR for my fix only, but he needed it anyway. I am now debating doing a PR in @cccs-jh's fork to have both the api.py fix plus the Stream fix, and maybe check what the random and int_convert fixes are doing from PR#43.

I see that you are doing plenty of work to move this project toward python3, and this is excellent news. I don't know how close you are to being done, and I understand that modifying the python2 code base while trying to migrate it may cause headaches, so I would be happy to know what your plans are. 

If pull requests for python2 are planned to be accepted it would really help if this PR and PR#43 could be merged so that all the fixes could be in one place. If python2 is essentially fixed until the python3 version is finished I can do pull requests on a fork and avoid the trouble of maintaining multiple pull requests with the same fixes on different versions.

I believe the fixes to the ADODB.Stream and other fixes in PR#43 would need to be applied to your python3/pyuno_elimination branches as well. Would having a PR into one of those branch make it easier to merge these PRs into the base master branch? I could also make changes in this PR if there is something keeping it from being merged. If the python2 version is fixed, would you be interested into having PR#43 merged into a python3 branch?

cccs-jh committed:
Fixing Now in python jit

Now is a python datetime object that when converted to string is not
valid python. Changing to_python to use repr for non-string non-VBAObject
native python objects so that they can be evaluated correctly.
Adding "import datetime" to the boilerplate to support datetime literals

cccs-kevin commented:
@kirk-sayre-work thoughts on this?

Commit:
filelen and fileread handling

cccs-ach committed:
filelen and fileread handling

cccs-ach committed:
firstindex field for regex.Execute()

cccs-ach committed:
byref fix + emulation

cccs-ach committed:
cleanup

cccs-ach committed:
cleanup

cccs-ach committed:
cleanup

cccs-ach committed:
coerce_to_ascii XOR

cccs-ach committed:
emulation fix

cccs-ach committed:
emulation fix

cccs-ach committed:
vm initialization

cccs-ach committed:
vm initialization

cccs-ach committed:
handle get read for documents

cccs-ach committed:
fix filelen error

cccs-ach committed:
get file read fix

cccs-ach committed:
read bug fix

Commit:
code style fix

cccs-ach committed:
Merge pull request #1 from cccs-jh/ach-wip

AL-2886

cccs-jh committed:
Hacky fix for lack of unicode support

Unicode characters can break parsing, removing them to avoid that.

cccs-jh committed:
Convert Context.report_action argument `action` to str

report_action can fail with a type error when the argument `action`
is not a string. Converting to string when `action` may not be a
string prevents this.
